src/components/timeline/form/TimelineEditorPanel.svelte

```diff
@@ -88,21 +91,24 @@
   let rows: Row[] = [];
   let timelines: Timeline[] = [];
   let verticalGuides: VerticalGuide[] = [];
+  let selectedTimeline: Timeline | undefined;
   let editorWidth: number;
 
-  $: rows = $selectedTimeline?.rows || [];
+  $: selectedTimeline = $view?.definition.plan.timelines.find(t => t.id === $selectedTimelineId);
+  $: rows = selectedTimeline?.rows || [];
   $: timelines = $view?.definition.plan.timelines || [];
-  $: verticalGuides = $selectedTimeline?.verticalGuides || [];
-  $: horizontalGuides = $selectedRow?.horizontalGuides || [];
-  $: yAxes = $selectedRow?.yAxes || [];
-  $: layers = $selectedRow?.layers || [];
-  $: rowHasActivityLayer = $selectedRow?.layers.find(isActivityLayer) || false;
+  $: verticalGuides = selectedTimeline?.verticalGuides || [];
+  $: selectedRow = rows.find(row => row.id === $selectedRowId);
```

Comment:
can we declare `selectedRow`, `yAxes`, `layers`, `rowHasActivityLayer`, `rowHasNonActivityChartLayer`, and `activityOptions` please?